src/main/scala/tech/cryptonomic/conseil/config/Security.scala

```diff
@@ -1,8 +1,38 @@
 package tech.cryptonomic.conseil.config
 
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.HttpRequest
+import akka.http.scaladsl.model.headers.RawHeader
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.Materializer
+import cats.effect.IO
+import cats.effect.concurrent.Ref
+import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport
 import pureconfig.generic.auto._
 
-object Security {
+import scala.concurrent.{ExecutionContext, Future}
+
+object Security extends ErrorAccumulatingCirceSupport {
+
+  /** Keys cache */
+  private val nautilusCloudKeys: Ref[IO, Set[String]] = Ref.of[IO, Set[String]](Set.empty).unsafeRunSync()
+
+  def updateKeys(
```

Comment:
Please add a doc string.

Additionally, it would be nice to get some logging from this function (or where ever appropriate) incase the key fetch from Nautilus cloud fails.

Reply:
added